tests/test_atom_client.py

```diff
@@ -0,0 +1,335 @@
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_add_child_resource_component.yaml'))
+def test_add_child_resource_component():
+    client = AtomClient(**AUTH)
+    slug = client.add_child('second-subfonds', title='Test child', level='item')
+    assert slug == 'test-child'
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_adding_child_with_note.yaml'))
+def test_adding_child_with_note():
+    client = AtomClient(**AUTH)
+    slug = client.add_child('test-fonds',
+                           title='Another subfonds',
+                           level='subfonds',
+                           notes=[{'type': 'general', 'content': 'This is a test note'}])
+    assert slug == 'another-subfonds'
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_posting_contentless_note.yaml'))
+def test_posting_contentless_note():
+    client = AtomClient(**AUTH)
+    slug = client.add_child('test-fonds',
+                           title='Yet another subfonds',
+                           level='subfonds',
+                           notes=[{'type': 'general', 'content': ''}])
+    assert client.get_record(slug)['notes'] == []
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_delete_record_resource.yaml'))
+def test_delete_record_resource():
+    client = AtomClient(**AUTH)
+    slug = 'another-subfonds'
+    assert client.get_record(slug)
+    r = client.delete_record(slug)
+    print str(r)
+    assert r['status'] == 'Deleted'
+    with pytest.raises(CommunicationError):
+        client.get_record(slug)
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_edit_archival_object.yaml'))
+def test_edit_archival_object():
+    client = AtomClient(**AUTH)
+    original = client.get_record('second-subfonds')
+    print str(original)
+    assert original['title'] == 'Second subfonds'
+    assert original['dates'][0]['end'] == '2015-01-01'
+    assert not original['notes']
+    new_record = {
+        'slug': 'second-subfonds',
+        'title': 'Test edited subfonds',
+        'start_date': '2014-11-01',
+        'end_date': '2015-11-01',
+        'date_expression': 'November, 2014 to November, 2015',
+        'notes': [{
+            'type': 'general',
+            'content': 'This is a test note'
+        }],
+    }
+    client.edit_record(new_record)
+    updated = client.get_record('second-subfonds')
+    assert updated['title'] == new_record['title']
+    assert updated['dates'][0]['begin'] == new_record['start_date']
+    assert updated['dates'][0]['end'] == new_record['end_date']
+    assert updated['dates'][0]['expression'] == new_record['date_expression']
+    assert updated['notes'][0]['type'] == new_record['notes'][0]['type']
+    assert updated['notes'][0]['content'] == new_record['notes'][0]['content']
+
+
+@vcr.use_cassette(os.path.join(THIS_DIR, 'fixtures/atom', 'test_edit_record_empty_note.yaml'))
+def test_edit_record_empty_note():
```

Comment:
I can't get this one to work. I am seeing this error: `PHP message: You cannot save an object that has been deleted` and I think it's coming from `informationobjectsUpdateAction.class.php`.